text/0000-stack-buildpacks.md

```diff
@@ -0,0 +1,668 @@
+
+[[entries]]
+name = "chromedriver"
+```
+
+If only the `chromedriver` is required, or the `google-chrome-stable` is already provided by the stack, the build phase would receive:
+
+```toml
+[[entries]]
+name = "chromedriver"
+```
+
+If the stack does not already provide `google-chrome-stable`, the stackpack may install it anyways. This prevents the case where an app may not work at runtime if that app requires only the `chromedriver` and the `google-chrome-stable` is not provided by the stack.
+
+## Caching and Restoring
+
+During the export phase, the lifecycle will store any excluded cached stackpack layers created during the build or extend phase in the cache.
+
+During the restore phase of the next build, the lifecycle will download the excluded cached stackpack layers with the cache and store them as tarballs in the `<layers>` directory and the `<run-layers>` directory (i.e. they will not be extracted). The `restorer` cannot extract the these layers because it will not run with root privileges. In addition, the `restorer` may run in a different container than the build, which means changes made to the base image are not guaranteed to carry forward. The `builder` will run with root privileges and untar these layers and cache, but will drop privileges after executing stackpacks and before running application buildpacks. The `extender` will also run with root privileges and untar the cache.
+
+During the build phase, after the stackpacks have run but before the application buildpacks have run, the lifecycle will forcefully delete everything in `[[excludes]]` after ALL stackpacks run, before buildpacks run (which implies that it does not need to do this for extend phase).
+
+## Rebasing an App
+
+Before an app image is rebased, the platform must re-run any stackpacks that were used to build the app image, against the new run-image. It will determine which stackpacks to run using a provided [`stack-group.toml`](https://github.com/buildpacks/spec/blob/main/platform.md#grouptoml-toml). The Buildpack Plan will be stored in a LABEL of the app image, which will be serialized back to a `plan.toml` during rebase. To each stackpack, it will pass the buildpack plan derived from the provided [`plan.toml`](https://github.com/buildpacks/spec/blob/main/platform.md#plantoml-toml).
+
+The image containing the stack buildpacks and the lifecycle binary must be provided to the rebaser operation as an argument. A platform may choose to provide the same stack buildpacks and lifecycle binary used during the build that create the app image being rebased, or it may provide updated versions (which may increase the risk of something failing in the rebase process).
+
+A platform may choose to store the stack buildpacks and extender binary in any of the following images:
+* A companion image of the app image
+* A builder image that is available when rebasing
+* The app image itself (store the stack buildpacks and builder binary alongside the application itself)
```

Comment:
Maybe we can just omit the options here and leave it as an exercise for the platform. We can describe the requirements (extend must run on the new run image with the stackpacks etc. etc.) without describing how the platform should do this (until we have more/better guidance). I worry that some options here are confusing or imply lifecycle requirements that we don't actually want to meet (e.g putting the stackpacks into the exported image).